apps/meteor/app/lib/server/methods/joinRoom.ts

```diff
@@ -26,6 +26,10 @@ Meteor.methods<ServerMethods>({
 			throw new Meteor.Error('error-invalid-room', 'Invalid room', { method: 'joinRoom' });
 		}
 
+		if (isOmnichannelRoom(room) && !room.open) {
```

Comment:
I'd say it would be better to put this validation inside of the `join` Room service method, since we will get rid of meteor methods soon and if this logic is being called from other places.